### PR TITLE
Adding documentation for overriding Angular style guide.

### PR DESCRIPTION
**Pull request by AbsentSemicolon**
This falls inline more what we want to see from Angular code.

### CODE REVIEW

File: lang/js/angular.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
...controller generate get github gives global gotosession gotosession; guide guide] have hint] how html https: identify if iife in included information inject injection instead into invoking is it johnpapa js jscs jshint just keeps like liked lines make makes manual manually master md#assigning-controllers md#bindable-members-up-top md#controlleras-with-vm md#js-hint members members] memory merged methods minified module more need needing needs ng-controller no not nothing of on one only or order outside path placement practice presented primary promises properties provides put readme refer refresh refresh; regards remove require resolve resolving restrict: return returning route rules same scope scope: search search; section section] sections service services sessions sessionscontroller should shouldn't since some sources stated strict strict` structure style superfluous supersedes supplies sure tempate template template: templateurl: the them, there they this those title to top topics true, truncated unless unneeded up us use using var variable variable1 variable1, variable2 variable2; variables view, vm vm] want waste way we we've what when which will with within without wrap your
```
Comment (fidian) [inline lang/js/angular.md]: Would you edit `bin/update-keywords` and change line 24 from this

    tr "A-Z \t\"()<>{}/=." "a-z\n" | sed -e "s/^$patt\\|$patt\$//g" | sort | uniq

to this

    tr "A-Z \t\"()<>{}/=.[];,:\`*" "a-z\n" | sed -e "s/^$patt\\|$patt\$//g" | sort | uniq

It adds several characters inside the search characters for `tr`. That will update a lot of these keywords and remove many, many duplicates. For instance, these changes will be made.

* "restrict:" -> "restrict" (clean up symbols)
* "strict strict'" -> "strict" (sorry, needed to use apostrophe here instead of backtick; cleans up symbols)
* "vm vm]" => "vm" (cleans up symbols)
* "variable1 variable1, variable2 variable2;" => "variable1 variable2" (cleans up symbols)

Not to mention all of those brackets at the beginning...

After that, run `bin/update-all-keywords` so the rest of the pages benefit.

File: lang/js/angular.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+---
+
+For this guide adhere to the [Angular 2 Style Guide] or the [Angular 1 Style Guide] unless stated below which supersedes those rules.
```
Comment (fidian) [inline lang/js/angular.md]: Comma after "guide" when phrased like what you have.

Comma before "which" or change "which" to "that".

Or reword.

> Unless stated otherwise below, adhere to the [Angular 2 Style Guide] or the [Angular 1 Style Guide].

File: lang/js/angular.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+Topics for this document are in the same order as they are in the [Angular 1 Style Guide].
+
+
+IIFE
+----
+
+We don't require the use of [IIFE] for every component. We compile all code into one file, remove all the `use strict` lines and put a `use strict` at the top of the minified file when compiled.
```
Comment (fidian) [inline lang/js/angular.md]: Instead of using "we" like this, could you reword it to be a third person form?

> [IIFE]s are not required for every component.  A build procedure shouls should combine files into one file, remove duplicate `"use strict";` lines are eliminated and finally minify the result. This is possible because no global variables are used within the Angular scope. It is bad practice to use global variables and nearly forbidden when developing an Angular application.
>
> Generate code without using superfluous [IIFE]s.

File: lang/js/angular.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+We can get away with this because we do not employ the use of global variables within the Angular scope. It is bad practice to use them, and will be avoided when developing an Angular application.
+
+Generate your code without using superfluous [IIFE]s.
+
+    // controller.js
+    "use strict";
+
+    angular.module("module").controller("SessionsController", function ($scope, dependency) {
```
Comment (fidian) [inline lang/js/angular.md]: Can we use ES6 like the JS style guide says?

    angular.module("module").controller("SessionController", ($scope, dependency) => {

File: lang/js/angular.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+        }
+
+        $scope.gotoSession = gotoSession;
+        $scope.refresh = refresh;
+        $scope.search = search;
+        variable1 = [];
+        variable2 = 'Sessions';
+    });
+
+If needing an activation function as can be the case when invoking a controller for a directive wrap everything within an `activate` function.
```
Comment (fidian) [inline lang/js/angular.md]: Comma after "directive" or reword by putting the "wrap everything" bit in front of the sentence.

File: lang/js/angular.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+            $scope.search = search;
+            $scope.sessions = [];
+            $scope.title = 'Sessions';
+        };
+    });
+
+
+Controllers
+-----------
+
```
Comment (fidian) [inline lang/js/angular.md]: You should have at least a line here explaining the topics that are covered under the Controllers heading. One should not have headings bumping into each other.

File: lang/js/angular.md [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+### controllerAs with vm
+
+Supersedes the topics presented in the [controllerAs with vm](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#controlleras-with-vm) section.
+
+We don't require the use of using a variable like `vm` within a controller, properties and methods can be assigned to `$scope` when they need to. Just make sure to only assigne to `$scope` what needs to be accessible.
```
Comment (fidian) [inline lang/js/angular.md]: I think you meant to say things can be assigned to `this`.

Typo: assigne

File: lang/js/angular.md [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+
+    <!-- games.html -->
+    <div ng-controller="GamesController as games">
+    </div>
+
+
+Services and Factories
+----------------------
+
+Services and Factories are formatted the same way and only return the functions outside sources should have access to.
```
Comment (fidian) [inline lang/js/angular.md]: Reword the preposition at the end of the sentence.

> ... only return the collection of functions that should be visible from outside.

File: lang/js/angular.md [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+Directives
+----------
+
+There is no need to format the directive by returning a variable. This is a waste of memory and lines as there is nothing being done with the variable except to return it.
+
+    // directive.js
+    "use strict";
+
+    function directive () {
+        return {
```
Comment (fidian) [inline lang/js/angular.md]: Both here and elsewhere: shouldn't we have jsdoc?

File: lang/js/angular.md [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+            templateUrl: "path/to/tempate.html"
+        }
+
+        angular.module("module").directive("directive", directive);
+    }
+
+
+Manual Annotating for Dependency Injection
+------------------------------------------
+
```
Comment (fidian) [inline lang/js/angular.md]: A sentence or two should go here. Don't forget the margin-top on headings is set to 2em.

File: lang/js/angular.md [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+### Manually Identify Dependencies
+
+The dependencies shouldn't use the `$inject` since we can inject the dependencies in the primary function when declaring the component.
```
Comment (fidian) [inline lang/js/angular.md]: Probably remove the first "the" in the sentence.

File: lang/js/angular.md [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+    // controller.js
+    "use strict";
+
+    angular.module("module").controller("SessionsController", function ($scope, dependency) {
+    // truncated to callout top section
+
+
+Resolving Promises
+------------------
+
```
Comment (fidian) [inline lang/js/angular.md]: Sentence here.

**Review by fidian — CHANGES_REQUESTED** [review]

File: lang/js/angular.md (hunk 11 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Fixed.

File: lang/js/angular.md (hunk 12 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Added sentence.

File: lang/js/angular.md (hunk 10 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Added sentence.

File: lang/js/angular.md (hunk 9 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Yes, I've added jsdoc where appropriate.

File: lang/js/angular.md (hunk 8 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Fixed.

File: lang/js/angular.md (hunk 7 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Fixed.

File: lang/js/angular.md (hunk 5 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Fixed.

File: lang/js/angular.md (hunk 6 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Added sentence.

File: lang/js/angular.md (hunk 4 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Fixed.

File: lang/js/angular.md (hunk 3 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Reworded.

File: lang/js/angular.md (hunk 2 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Added commas.

File: lang/js/angular.md (hunk 1 above)
Comment (AbsentSemicolon) [reply lang/js/angular.md]: Fixed and updated.

Comment (fidian) [reply lang/js/angular.md]: Thanks, looks good.

File: lang/js/angular.md (hunk 2 above)
Comment (fidian) [reply lang/js/angular.md]: Resolved.

File: lang/js/angular.md (hunk 3 above)
Comment (fidian) [reply lang/js/angular.md]: Resolved.

File: lang/js/angular.md (hunk 5 above)
Comment (fidian) [reply lang/js/angular.md]: Resolved.

File: lang/js/angular.md (hunk 6 above)
Comment (fidian) [reply lang/js/angular.md]: Resolved

File: lang/js/angular.md (hunk 7 above)
Comment (fidian) [reply lang/js/angular.md]: Resolved.

File: lang/js/angular.md (hunk 8 above)
Comment (fidian) [reply lang/js/angular.md]: Resolved.

File: lang/js/angular.md (hunk 9 above)
Comment (fidian) [reply lang/js/angular.md]: Resolved.

File: lang/js/angular.md (hunk 10 above)
Comment (fidian) [reply lang/js/angular.md]: Resolved.

File: lang/js/angular.md (hunk 11 above)
Comment (fidian) [reply lang/js/angular.md]: Resolved.

File: lang/js/angular.md (hunk 12 above)
Comment (fidian) [reply lang/js/angular.md]: Resolved.

**Review by fidian — APPROVED** [review]
